### PR TITLE
fix for https://github.com/containrrr/shoutrrr/issues/70

### PR DESCRIPTION
**Pull request by ellisab**
Signed-off-by: Alexandru Bonini <boninialexandru@gmail.com>

### CODE REVIEW

**Review by piksel — APPROVED** [review]

File: pkg/services/rocketchat/rocketchat_config.go [hunk 1]
```diff
@@ -46,7 +47,9 @@ func (config *Config) SetURL(serviceURL *url.URL) error {
 	config.TokenA = path[1]
 	config.TokenB = path[2]
 	if len(path) > 3 {
-		if path[3][0:1] != "@" {
+		if serviceURL.Fragment != "" {
+			config.Channel = "#" + strings.TrimLeft(serviceURL.Fragment, "#")
```
Comment (simskij) [inline pkg/services/rocketchat/rocketchat_config.go:51]: https://golang.org/pkg/strings/#TrimLeft

> To remove a prefix, use TrimPrefix instead.

No biggie here as the behavior will be the same, but to keep intentions clear, I think we should switch to `TrimPrefix` anyway.

File: pkg/services/rocketchat/rocketchat_config.go [hunk 2]
```diff
@@ -46,7 +47,9 @@ func (config *Config) SetURL(serviceURL *url.URL) error {
...
+		} else if path[3][0:1] != "@" {
```
Comment (simskij) [inline pkg/services/rocketchat/rocketchat_config.go:52]:
```suggestion
		} else if !strings.HasPrefix(path[3], "@") {
```

**Review by simskij — CHANGES_REQUESTED** [review]

**Comment by simskij** [comment]
Thank you for your excellent contribution! Some minor nitpicks above, and then I think we're good to go 👍

File: pkg/services/rocketchat/rocketchat_config.go (hunk 2 above)
Comment (ellisab) [reply pkg/services/rocketchat/rocketchat_config.go]: Implemented the suggested changes

File: pkg/services/rocketchat/rocketchat_config.go (hunk 1 above)
Comment (ellisab) [reply pkg/services/rocketchat/rocketchat_config.go]: Implemented the suggested change

**Comment by simskij** [comment]
@all-contributors add @ellisab for code

**Pull request merged**